File: langgraph/data/equiv/140.py
def adjust_whitespace(input_string):
    result = ""
    index = 0
    begin, finish = 0, 0
    while index < len(input_string):
        if input_string[index] == " ":
            finish += 1
        else:
            space_count = finish - begin
            if space_count > 2:
                result += "-" + input_string[index]
            elif space_count > 0:
                result += "_" * space_count + input_string[index]
            else:
                result += input_string[index]
            begin, finish = index + 1, index + 1
        index += 1
    final_space_count = finish - begin
    if final_space_count > 2:
        result += "-"
    elif final_space_count > 0:
        result += "_"
    return result

# method name: fix_spaces -> adjust_whitespace
# parameter name: text -> input_string
# variable name: new_text -> result, i -> index, start -> begin, end -> finish
# replaced inline conditions with a separate variable for space count (space_count)
# maintained the same logic and output structure

File: langgraph/result/qwen/3b-eq2js/140.js
function adjustWhitespace(inputString) {
    let result = "";
    let index = 0;
    let begin = 0;
    let finish = 0;
    while (index < inputString.length) {
        if (inputString[index] === " ") {
            finish++;
        } else {
            const spaceCount = finish - begin;
            if (spaceCount > 2) {
                result += "-" + inputString[index];
            } else if (spaceCount > 0) {
                result += "_".repeat(spaceCount) + inputString[index];
            } else {
                result += inputString[index];
            }
            begin = index + 1;
            finish = index + 1;
        }
        index++;
    }
    const finalSpaceCount = finish - begin;
    if (finalSpaceCount > 2) {
        result += "-";
    } else if (finalSpaceCount > 0) {
        result += "_";
    }
    return result;
}